pkg/watch/restricted_enqueue_request_for_owner.go

```diff
@@ -1,227 +0,0 @@
-	if len(kinds) != 1 {
-		err = fmt.Errorf("%w: expected exactly 1 kind for OwnerType %T, but found %s kinds",
-			ErrNoUniqueKind, e.OwnerType, kinds)
-		e.Log.Error(nil, "expected exactly 1 kind for OwnerType", "owner type",
-			fmt.Sprintf("%T", e.OwnerType), "kinds", kinds)
-		return err
-	}
-	// Cache the Group and Kind for the OwnerType
-	e.groupKind = schema.GroupKind{Group: kinds[0].Group, Kind: kinds[0].Kind}
-	return nil
-}
-
-// getOwnerReconcileRequest looks at object and builds a map of reconcile.Request to reconcile
-// owners of object that match e.OwnerType.
-func (e *RestrictedEnqueueRequestForOwner) getOwnerReconcileRequest(
-	oldIfAny, object client.Object, result map[reconcile.Request]any,
-) {
-	// Iterate through the OwnerReferences looking for a match on Group and Kind against what was requested
-	// by the user
-	refs := e.getOwnersReferences(object)
-	for _, ref := range refs {
-		// Parse the Group out of the OwnerReference to compare it to what was parsed out of the requested OwnerType
-		refGV, err := schema.ParseGroupVersion(ref.APIVersion)
-		if err != nil {
-			e.Log.Error(err, "Could not parse OwnerReference APIVersion",
-				"api version", ref.APIVersion)
-			return
-		}
-		e.getOwnerReconcileRequestFromOwnerReference(oldIfAny, object, result, ref, refGV)
-	}
-}
-
-func (e *RestrictedEnqueueRequestForOwner) getOwnerReconcileRequestFromOwnerReference(
-	oldIfAny, object client.Object,
-	result map[reconcile.Request]any,
-	ref apimetav1.OwnerReference,
-	refGV schema.GroupVersion,
-) {
-	// Compare the OwnerReference Group and Kind against the OwnerType Group and Kind specified by the user.
-	// If the two match, create a Request for the objected referred to by
-	// the OwnerReference.  Use the Name from the OwnerReference and the Namespace from the
-	// object in the event.
-	if ref.Kind != e.groupKind.Kind || refGV.Group != e.groupKind.Group {
-		return
-	}
-
-	// Match found - add a Request for the object referred to in the OwnerReference
-	request := reconcile.Request{NamespacedName: types.NamespacedName{
-		Name: ref.Name,
-	}}
-
-	// if owner is not namespaced then we should set the namespace to the empty
-	mapping, err := e.mapper.RESTMapping(e.groupKind, refGV.Version)
-	if err != nil {
-		e.Log.Error(err, "Could not retrieve rest mapping", "kind", e.groupKind)
-		return
-	}
-	if mapping.Scope.Name() != meta.RESTScopeNameRoot {
-		request.Namespace = object.GetNamespace()
-	}
-
-	if oldIfAny != nil {
-		componentOld, _ := machineryruntime.DefaultUnstructuredConverter.ToUnstructured(oldIfAny)
-		componentNew, _ := machineryruntime.DefaultUnstructuredConverter.ToUnstructured(object)
-		oldState, okOld, _ := unstructured.NestedString(componentOld, "status", "state")
-		newState, okNew, _ := unstructured.NestedString(componentNew, "status", "state")
-
-		if err != nil || !okNew || !okOld {
-			e.Log.Error(err, "error getting owner")
-		}
-
-		if oldState != newState {
```

Comment:
Do you think it's fine to use the default implementation that does not check state changes and simply enqueues a reconciliation request based on the ownership relationships. I mean to declare the state change detection as not necessary?
I don't have enough insight to judge if this will result in more enqueues or not.

Reply:
yes, I explained here:
> The reason why we previously introduced this customized RestrictedEnqueueRequestForOwner is because we want to limit kyma CR enqueue only triggered by update event which makes manifest CR state changes (we use SSA, which update manifest very frequently before). However, since we limited the SSA in this https://github.com/kyma-project/lifecycle-manager/issues/1456, there is not necessary to have this customized version, we can directly use the native EnqueueRequestForOwner to achieve the same goal.

Reply:
Ok, missed that the first time. Thanks!